src/java.desktop/macosx/native/libawt_lwawt/awt/ApplicationDelegate.m

```diff
@@ -409,6 +410,14 @@ - (NSApplicationTerminateReply)applicationShouldTerminate:(NSApplication *)app {
     return NSTerminateLater;
 }
 
+- (BOOL)applicationSupportsSecureRestorableState:(NSApplication *)app {
+    BOOL supportsSecureState = YES;
+    if (getenv("AWT_DISABLE_NSDELEGATE_SECURE_SAVE") != NULL) {
+       supportsSecureState = NO;
+    }
```

Comment:
I think that (and in the other case), it is best to read the env. var only once, not keep checking it on ever call to this method. Theoretically if someone used putenv/setenv (yes they'd need native code), it would cause us to change the answer.

Yes, this makes the code a bit more complicated but it will be safer.
maybe you could maybe use static locals
(BOOL)applicationSupportsSecureRestorableState:(NSApplication *)app {
     static BOOL checked = NO;
     static BOOL supportsSecureState = YES;
     if (checked == NO) {
        checked = YES;
        if (getenv("AWT_DISABLE_NSDELEGATE_SECURE_SAVE") != NULL) {
            supportsSecureState = NO;
     }
     return supportsSecureState;
}

Reply:
@prrace I think using static variables to read env variables only once is a good idea since I see `applicationSupportsSecureRestorableState()` being called every time the window loses focus.